api/pb/types.pb.go

```diff
@@ -55,19 +55,15 @@ func (Pin_PinType) EnumDescriptor() ([]byte, []int) {
 }
```

Comment:
This is an auto-generated file which we are not suppose to edit.
Instead of adding it into our source code, shouldn't we add the logic to generate the file into `make build` and let it auto-generate everytime?

Reply:
You have a point, but there are some benefits in committing the generated file,

* Generation process won't error or produce different results if something changed in the generation tool during builds
* Users don't need to have the protobuf toolchain installed to build cluster
* Users can bind against cluster libraries without having to "build" cluster first.

Committing the generated protobuf is the rule in our ecosystem afaik.